cache-url/spec/AmpUrlSpec.js

```diff
@@ -0,0 +1,54 @@
+/**
+ * Copyright 2018 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+'use strict';
+
+const AmpUrl = require('../lib/AmpUrl');
+
+describe('AmpUrl', () => {
+  const ampUrl = new AmpUrl();
+  const cache = {
+    updateCacheApiDomainSuffix: 'cdn.ampproject.org'
+  };
+
+  describe('cacheUrl', () => {
+    const tests = [
+      {canonical: 'https://www.example.com',
```

Comment:
Can you fix the formatting to make this easier to read? 

```
{
  canonical: 'http://www.example.com/index.html',
  cache: 'https://www-example-com.cdn.ampproject.org/c/www.example.com/index.html'
},
```

Reply:
Also, I don't think we should name this url instead of canonical.

Reply:
I think I would've simplified this to:

```
{
  'http://www.example.com/index.html': 'https://www-example-com.cdn.ampproject.org/c/www.example.com/index.html',
'http://www.example.com/index.html': 'https://www-example-com.cdn.ampproject.org/c/www.example.com/index.html',
...
},

Reply:
Fixed the formatting. I'm keeping the object, as it makes easier to use later in the code.